### ISSUE TITLE
aggregate data holding by GB instead of count

### ISSUE THREAD

**Issue by rabernat** [pangeo-forge/cmip6-pipeline#16]
This fantastic notebook developed by @aradhakrishnanGFDL and @naomi-henderson is super useful for understanding what data we have in the cloud: https://github.com/aradhakrishnanGFDL/gfdl-aws-analysis/blob/master/examples/a1r-CompareAWS-NCvsZarr.ipynb

However, it's aggregated by number of datasets:
![image](https://user-images.githubusercontent.com/1197350/112692678-b41b8980-8e55-11eb-8f9d-158e4b17d091.png)

Once #15 is implemented, it should be easy to get the size of each dataset. So we can aggregate by summing the sizes, rather than just counting datasets.

This is required for the AWS folks to move forward in allocating more storage space for us. We need the numbers in GB.

**Comment by aradhakrishnanGFDL** [comment]
@rabernat @naomi-henderson I have updated the [notebook](https://github.com/aradhakrishnanGFDL/gfdl-aws-analysis/blob/master/examples/a1r-CompareAWS-NCvsZarr.ipynb) to read the parquet files for esgf-world and reflect the data volume in the visualizations, along with some notes. I am still checking couple of things on esgf-world to make sure the volumes reported tally across.  zarr size is not reflected in the charts yet, shows 0 for now. 

After the inventory/parquet files are available for cmip6-pds, I could plug in those numbers to the notebook if you'd like. Please note that parquet files should be public or readable from a different account if I need to run it from my end.

**Comment by rabernat** [comment]
This is great! Pasting a figure here for documenation

![image](https://user-images.githubusercontent.com/1197350/113182476-f1e72c00-9220-11eb-8361-a20bfa53953f.png)

Since all the numbers are in the 1000s, maybe we should use TB as our units?

> After the inventory/parquet files are available for cmip6-pds, I could plug in those numbers to the notebook if you'd like.

To be clear, I am asking if you (Aparna) can generate the inventory files for `cmip6-pds`. Since you have already figured out how to use the tool, I think that is by far the most efficient path. Naomi has sent you the credentials.

**Comment by aradhakrishnanGFDL** [comment]
@rabernat Yes, understood! Configured. Will provide updates in #15 when a parquet file is generated so I know the configuration was correctly done.

**Comment by aradhakrishnanGFDL** [comment]
@rabernat @naomi-henderson 
I've updated the notebook to include the data volumes. Does it look reasonable? 
https://github.com/aradhakrishnanGFDL/gfdl-aws-analysis/blob/master/examples/a1r-CompareAWS-NCvsZarr.ipynb

**Comment by rabernat** [comment]
The numbers look reasonable if they are GB. The legend says TB, but the values are O(100000).

**Comment by aradhakrishnanGFDL** [comment]
oops, strike that, I will push the correct update soon.

**Comment by aradhakrishnanGFDL** [comment]
notebook updated 
https://github.com/aradhakrishnanGFDL/gfdl-aws-analysis/blob/master/examples/a1r-CompareAWS-NCvsZarr.ipynb

**Comment by naomi-henderson** [comment]
Yes, @aradhakrishnanGFDL , the notebook looks reasonable.  I love your Venn diagram at the end!

It probably doesn't make much difference, but it looks like the zarr parquet file is a listing of the sizes of all of the `variable_id` chunks.  This is close enough for this purpose, but not really quite right, since the datasets (and their netcdf files) often contain auxiliary variables (such as `ps` along with the atmosphere data on hybrid vertical coordinates).  
Instead, it might be easier to find the paths of all `.zmetadata` files because there is one such file for each zarr store.  Then the total size of the containing directory is the size of the dataset.  In fact, your parquet file will be much, much smaller!

**Comment by aradhakrishnanGFDL** [comment]
Thanks @naomi-henderson! The parquet file is from the S3 inventory and seems to be at an object granularity, so it was hard to work with it initially as implied.

I added an extra column (vstore) to the parquet..and then grouped by vstore, aggregating by sum of sizes. vstore contains .zmetadata that you refer to, so I think we are okay? Please let me know what you think and I'll revise the code. 

Verifying..

t=zarr_inventory_dd.loc[zarr_inventory_dd['vstore']=='CMIP6/AerChemMIP/BCC/BCC-ESM1/ssp370/r1i1p1f1/Amon/rsds/gn/v20190624']['size'].head()
returns 11777175. The same being returned via aws s3 ls --recursive. 

One thing I do hope to get to soon.. is to substitute with the most recent inventory file (zarr).
The inventory file currently accounts for 814TB in cmip6-pds, but there may be 40TB more in the recent one, cross-checking S3 Storage Lens from your AWS account. 
If there is a more recent cmip6-pds intake-esm catalog, I can also point to that. Please let me know. Thanks,

**Comment by naomi-henderson** [comment]
@aradhakrishnanGFDL , it looks like you are using an old csv file - used just for the particular comparisons we were doing at the time (that is why is has `/compare` in the url).  

The continually updating csv file is `s3://cmip6-pds/pangeo-cmip6-noQC.csv.gz`
